### PR TITLE
Use array access, and remove unnecessary double `error` from array lookup

### PR DESCRIPTION
**Pull request by WyriHaximus**

### CODE REVIEW

File: src/ChildProcess/Adapter.php [hunk 1]
```diff
@@ -130,8 +130,8 @@ public function callFilesystem($function, $args, $errorResultCode = -1)
     {
         return $this->pool->rpc(Factory::rpc($function, $args))->then(function (Payload $payload) {
             return \React\Promise\resolve($payload->getPayload());
-        }, function (Payload $payload) {
-            return \React\Promise\reject(new Exception($payload->getPayload()['error']['error']['message']));
+        }, function ($payload) {
+            return \React\Promise\reject(new Exception($payload['error']['message']));
```
Comment (clue) [inline src/ChildProcess/Adapter.php]: The title suggests there must be a way to verify this is "the correct way"? :shipit:

Comment (WyriHaximus) [reply src/ChildProcess/Adapter.php]: Yeah using array access and ditching the extra `error` index solve the errors.

(Also updated the title.)

Comment (clue) [reply src/ChildProcess/Adapter.php]: Is this tested or can we add some tests for this? :+1:

Comment (WyriHaximus) [reply src/ChildProcess/Adapter.php]: Added a test for this :+1:

**Pull request merged**